api/routes/auth.go

```diff
@@ -222,6 +225,74 @@ func (h *Handler) AuthGetToken(c gateway.Context) error {
 	return c.JSON(http.StatusOK, res)
 }
 
+func (h *Handler) AuthCustomToken(c gateway.Context) error {
+	var req requests.AuthCustomToken
+	if err := c.Bind(&req); err != nil {
+		return err
+	}
+
+	if err := c.Validate(&req); err != nil {
+		return err
+	}
+
+	var userID string
+	if v := c.ID(); v != nil {
+		userID = v.ID
+	}
+
+	token, err := h.service.AuthCustomToken(c.Ctx(), userID, req.Name, req.ExpiresAt)
+	if err != nil {
+		return err
+	}
+
+	return c.JSON(http.StatusOK, map[string]interface{}{
+		"token": token["token"],
+	})
+}
+
+func (h *Handler) AuthGetCustomToken(c gateway.Context) error {
+	var req requests.UserParam
+
+	if err := c.Bind(&req); err != nil {
+		return err
+	}
+
+	if err := c.Validate(&req); err != nil {
+		return err
+	}
+
+	res, err := h.service.AuthGetCustomToken(c.Ctx(), req.ID)
+	if err != nil {
+		return err
+	}
+
+	return c.JSON(http.StatusOK, res)
+}
+
+func (h *Handler) AuthRemoveCustomToken(c gateway.Context) error {
+	var req requests.UserParam
+
+	if err := c.Bind(&req); err != nil {
+		return err
+	}
+
+	if err := c.Validate(&req); err != nil {
+		return err
+	}
+
+	token, err := h.service.AuthGetCustomTokenByID(c.Ctx(), req.ID)
+	if err != nil {
+		return err
+	}
+
+	err = h.service.AuthRemoveCustomToken(c.Ctx(), token.ID, token.Token)
+	if err != nil {
+		return err
+	}
```

Comment:
I believe we should search for the token in the `AuthRemoveCustomToken` service instead of the route. In my opinion, this approach prevents leaking out the business rules.

Reply:
changed the search for token to service file